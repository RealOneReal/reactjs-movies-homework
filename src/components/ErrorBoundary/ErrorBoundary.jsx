import React, { Component } from "react";
import { Box, Container, Typography } from "@mui/material";

class ErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  render() {
    if (this.state.hasError) {
      return (
        <Box sx={{ backgroundColor: "secondary.dark" }}>
          <Container maxWidth="lg" sx={{ pt: 2 }}>
            <Box sx={{ textAlign: "center", height: "90vh" }}>
              <Typography variant="h4" color="primary.dark">
                Something went wrong ...
              </Typography>
            </Box>
          </Container>
        </Box>
      );
    }
    return this.props.children;
  }
}

export default ErrorBoundary;
