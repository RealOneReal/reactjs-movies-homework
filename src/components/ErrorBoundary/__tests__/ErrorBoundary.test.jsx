import { render } from "@testing-library/react";
import ErrorBoundary from "../ErrorBoundary";

const mockComponent = <div data-testid="test">Hello</div>;
const MockErrorComponent = () => {
  throw Error("Boom!");
};
// window.console = { error: jest.fn() };

describe("ErrorBoundary", () => {
  it("should render children component without error", () => {
    const { container } = render(
      <ErrorBoundary>{mockComponent}</ErrorBoundary>
    );
    expect(container).toMatchSnapshot();
  });
  it("should render saved JSX from ErrorBoundary ", () => {
    const { container } = render(
      <ErrorBoundary>
        <MockErrorComponent />
      </ErrorBoundary>
    );

    expect(container).toMatchSnapshot();
  });
});
