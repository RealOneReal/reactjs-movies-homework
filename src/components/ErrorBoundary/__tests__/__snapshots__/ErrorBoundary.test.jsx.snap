// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render children component without error 1`] = `
<div>
  <div
    data-testid="test"
  >
    Hello
  </div>
</div>
`;

exports[`ErrorBoundary should render saved JSX from ErrorBoundary  1`] = `
<div>
  <div
    class="MuiBox-root css-4ozt90"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-r5r38p-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-l3vw6"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-1tg5hpw-MuiTypography-root"
        >
          Something went wrong ...
        </h4>
      </div>
    </div>
  </div>
</div>
`;
